src/main/java/hudson/plugins/logparser/LogParserPublisher.java

```diff
@@ -19,18 +21,21 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import jenkins.tasks.SimpleBuildStep;
 import net.sf.json.JSONObject;
 
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 import org.kohsuke.stapler.StaplerRequest;
 
-public class LogParserPublisher extends Recorder implements Serializable {
+public class LogParserPublisher extends Recorder implements SimpleBuildStep, Serializable {
     private static final long serialVersionUID = 1L;
-    public final boolean unstableOnWarning;
-    public final boolean failBuildOnError;
-    public final boolean showGraphs;
-    public final String parsingRulesPath;
-    public final boolean useProjectRule;
-    public final String projectRulePath;
+    public boolean unstableOnWarning;
+    public boolean failBuildOnError;
+    public boolean showGraphs;
+    public String parsingRulesPath;

```

Comment:
@oleg-nenashev sure, I get the problem, but I think it needs to be handle on a case-by-case basis i.e. they're not always evil and in this case I'm not sure that they are. If a particular property needs to be immutable then that could be handled in the setter for that property. There are tradeoffs in both cases (`@DataBoundSetter` Vs `@DataBoundConstructor`).